src/uri.rs

```diff
@@ -102,86 +112,90 @@ impl FromStr for UUri {
     /// let uri_from = UUri::from_str("//VIN.vehicles/A8000/2/1A50").unwrap();
     /// assert_eq!(uri, uri_from);
     /// ````
+    // [impl->dsn~authority-name-length~1]
+    // [impl->dsn~uri-scheme~1]
+    // [impl->dsn~uri-host-only~1]
+    // [impl->dsn~uri-authority-mapping~1]
+    // [impl->dsn~uri-path-mapping~1]
+    // [impl->req~uri-serialization~1]
     fn from_str(uri: &str) -> Result<Self, Self::Err> {
         if uri.is_empty() {
             return Err(UUriError::serialization_error("URI is empty"));
         }
+        let parsed_uri = URIReference::try_from(uri)
```

Comment:
Could uriparse deal with UURI w/o authority?
For example: https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uri.adoc#31-examples

Reply:
yes, that is one of the test cases https://github.com/eclipse-uprotocol/up-rust/blob/01e3149aa7ebffeb81f03746791693f198076c5e/src/uri.rs#L549